src/balena-init.sql

```diff
@@ -12,86 +12,129 @@ ADD COLUMN IF NOT EXISTS "jwt secret" VARCHAR(255) NULL;
+CREATE INDEX IF NOT EXISTS "device_should_be_running_release_idx"
+ON "device" ("should be running-release");
+-- Also optimizes the rule `It is necessary that each device that should be
 -- managed by a supervisor release, should be managed by a supervisor release
 -- that is for a device type that describes the device.`
 -- as it relies heavily on these two columns
 CREATE INDEX IF NOT EXISTS "device_supervisor_release_device_type_idx"
 ON "device" ("should be managed by-supervisor release", "is of-device type");
 
+-- "device config variable"."device" is the first part of an automated unique index
+
+-- "device environment variable"."device" is the first part of an automated unique index
+
+-- "device tag"."device" is the first part of an automated unique index
+
+-- "device service environment variable"."service install" is the first part of an automated unique index
+
+-- "gateway download"."image" is the first part of an automated unique index
+CREATE INDEX IF NOT EXISTS "gateway_download_device_idx"
+ON "gateway download" ("is downloaded by-device");
+
+CREATE INDEX IF NOT EXISTS "image_is_build_of_service_idx"
+ON "image" ("is a build of-service");
+
+-- "image environment variable"."release image" is the first part of an automated unique index
+
 CREATE INDEX IF NOT EXISTS "ii_ipr_idx"
 ON "image install" ("is provided by-release");
-
--- Optimisation for device state query
+-- Also optimizes for device state query
 CREATE INDEX IF NOT EXISTS "image_install_image_device_idx"
 ON "image install" ("installs-image", "device");
-
--- Optimisation for device state query
+-- Also optimizes for device state query
 CREATE INDEX IF NOT EXISTS "image_install_device_image_idx"
 ON "image install" ("device", "installs-image");
 
--- Optimisation for device state query
+-- "image-is part of-release"."image" is the first part of an automated unique index
+-- Also optimizes device state query
 CREATE INDEX IF NOT EXISTS "ipr_ipr_image_idx"
 ON "image-is part of-release" ("is part of-release", "image");
 
--- Optimisation for device state query
+-- "image label"."release image" is the first part of an automated unique index
+
+-- TODO: Check what the extra columns are optimizing for
+CREATE INDEX IF NOT EXISTS "release_application_commit_status_idx"
+ON "release" ("belongs to-application", "commit", "status");
+
+CREATE INDEX IF NOT EXISTS "release_tag_release_idx"
+ON "release tag" ("release");
+
+-- "service"."application" is the first part of an automated unique index
+
+-- "service environment variable"."service" is the first part of an automated unique index
+
+-- "service label"."service" is the first part of an automated unique index
+
+-- "service install"."device" is the first part of an automated unique index
+CREATE INDEX IF NOT EXISTS "service_install_service_idx"
+ON "service install" ("installs-service");
+
+CREATE INDEX IF NOT EXISTS "supervisor_release_device_type_idx"
+ON "supervisor release" ("is for-device type");
```

Comment:
Any reason these two are not in the migration?
For sure they are not in my BoB's DB.

Reply:
I must have missed them when comparing what was moved vs new - fixed